### PR TITLE
Remove types for TS <=3.9

### PR DESCRIPTION
**Pull request by jablko**
I removed the `ts3.9/` directories and `typesVersions["<=3.9"]` fields because [this assertion](https://github.com/microsoft/DefinitelyTyped-tools/blob/dae5c7684839a3bd290a3029ba3fd72a19d3c63a/packages/dtslint/src/index.ts#L197) became false when [support for TS 3.9 ended](https://github.com/microsoft/DefinitelyTyped-tools/pull/473):
```sh
npm test call-bind

dtslint@0.0.115
AssertionError [ERR_ASSERTION]: Expected "actual" to be strictly unequal to: -1
    at next (DefinitelyTyped/node_modules/@definitelytyped/dtslint/dist/index.js:185:12)
    at Array.map (<anonymous>)
    at runTests (DefinitelyTyped/node_modules/@definitelytyped/dtslint/dist/index.js:158:88)
    at async main (DefinitelyTyped/node_modules/@definitelytyped/dtslint/dist/index.js:79:9)

### CODE REVIEW

File: types/call-bind/package.json [hunk 1]
```diff
@@ -1,7 +0,0 @@
-{
-    "private": true,
-    "types": "index",
-    "typesVersions": {
-        "<=3.9": { "*": ["ts3.9/*"] }
```
Comment (ljharb) [inline types/call-bind/package.json:5]: i don’t think deleting the types entirely is the proper approach

File: types/function-bind/package.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{
-    "private": true,
-    "types": "index",
-    "typesVersions": {
-        "<=3.9": {
-            "*": ["ts3.9/*"]
```
Comment (ljharb) [inline types/function-bind/package.json:6]: same here

**Review by ljharb — CHANGES_REQUESTED** [review]

File: types/call-bind/package.json (hunk 1 above)
Comment (jablko) [reply types/call-bind/package.json:5]: @ljharb For newer versions of TS (>3.9) the types are still there and still work after this change: I didn't remove [`types/call-bind/index.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/887ff7fb0eab64c6437e88aed4f1160b5d8ffde8/types/call-bind/index.d.ts), [`types/function-bind/index.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/887ff7fb0eab64c6437e88aed4f1160b5d8ffde8/types/function-bind/index.d.ts), etc.

For TS <=3.9 you can continue use the last version before support ended, using the e.g. ts3.9 npm tag:
`npm install call-bind@ts3.9` https://github.com/DefinitelyTyped/DefinitelyTyped/blob/09dc238010a270259288fe4441f0bd2eda7651db/README.md?plain=1#L89

This is standard when support for older versions of TS ends, I think. Here's an analogous example of removing types for TS <=3.7: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57450#issue-1067743511

Comment (ljharb) [reply types/call-bind/package.json:5]: I don't care about TS 3.9; i just want to make sure these packages' types still work in modern TS. Is the package.json not required for that?

Comment (MysteryBlokHed) [reply types/call-bind/package.json:5]: The package.json should only be required if it specifies external dependencies afaik

**Review by ljharb — APPROVED** [review]
call-bind and function-bind LGTM

**Review by atwwei — APPROVED** [review]

File: types/call-bind/package.json (hunk 1 above)
Comment (sandersn) [reply types/call-bind/package.json:5]: That's right. The `@types` publisher creates a package.json when one is not present in the repo.

**Review by sandersn — APPROVED** [review]

**Pull request merged**